hack/e2e-dynamicfile.sh

```diff
@@ -54,6 +55,75 @@ function e2e_mountfile() {
 
 }
 
+function e2e_dynamicfile_username_prefix_enforce(){
+  set +e
+  RoleOutput=$(aws iam get-role --role-name ${USERNAME_TEST_ROLE} 2>/dev/null)
+
+  if [ -z "$RoleOutput" ]; then
+      sed -e "s|{{AWS_ACCOUNT}}|${AWS_ACCOUNT}|g" \
+              "${policies_template}" > "${policies_json}"
+      sleep 2
+      aws iam create-role --role-name ${USERNAME_TEST_ROLE} --assume-role-policy-document file://${policies_json} 1>/dev/null
+      sleep 10
+  fi
+
+  #detect if run on github and allow the test account to assume role accordingly
+  if [ $CI = true ]
+      then
+          OUT=$(aws iam list-attached-user-policies --user-name awstester)
```

Comment:
Are we using an IAM user for these tests? Can we update the test infra to use [GitHub's OIDC auth](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) instead?

Reply:
Sync'd offline, this will be a separate PR to test-infra